components/dashboard/src/projects/ProjectVariables.tsx

```diff
@@ -0,0 +1,126 @@
+/**
+ * Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+ * Licensed under the GNU Affero General Public License (AGPL).
+ * See License-AGPL.txt in the project root for license information.
+ */
+
+import { Project, ProjectEnvVar } from "@gitpod/gitpod-protocol";
+import { useContext, useEffect, useState } from "react";
+import CheckBox from "../components/CheckBox";
+import InfoBox from "../components/InfoBox";
+import { Item, ItemField, ItemsList } from "../components/ItemsList";
+import Modal from "../components/Modal";
+import { getGitpodService } from "../service/service";
+import { ProjectContext } from "./project-context";
+import { ProjectSettingsPage } from "./ProjectSettings";
+
+export default function () {
+    const { project } = useContext(ProjectContext);
+    const [ envVars, setEnvVars ] = useState<ProjectEnvVar[]>([]);
+    const [ showAddVariableModal, setShowAddVariableModal ] = useState<boolean>(false);
+
+    const updateEnvVars = async () => {
+        if (!project) {
+            return;
+        }
+        const vars = await getGitpodService().server.getProjectEnvironmentVariables(project.id);
+        const sortedVars = vars.sort((a, b) => a.name.toLowerCase() > b.name.toLowerCase() ? 1 : -1);
+        setEnvVars(sortedVars);
+    }
+
+    useEffect(() => {
+        updateEnvVars();
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [project]);
+
+    const deleteEnvVar = async (variableId: string) => {
+        await getGitpodService().server.deleteProjectEnvironmentVariable(variableId);
+        updateEnvVars();
+    }
+
+    return <ProjectSettingsPage project={project}>
+        {showAddVariableModal && <AddVariableModal project={project} onClose={() => { updateEnvVars(); setShowAddVariableModal(false); }} />}
+        <div className="mb-2 flex">
+            <div className="flex-grow">
+                <h3>Project Variables</h3>
+                <h2 className="text-gray-500">Manage environment variables for your project.</h2>
+            </div>
+            {envVars.length > 0 && <button onClick={() => setShowAddVariableModal(true)}>Add Variable</button>}
+        </div>
+        {envVars.length === 0
+            ? <div className="bg-gray-100 dark:bg-gray-800 rounded-xl w-full py-28 flex flex-col items-center">
+                <h3 className="text-center pb-3 text-gray-500 dark:text-gray-400">No Environment Variables</h3>
+                <button onClick={() => setShowAddVariableModal(true)}>New Variable</button>
```

Comment:
**thought**: It could be interesting to surface inherited user-specific environment variables below project-specific environment variables. Added follow-up https://github.com/gitpod-io/gitpod/issues/7532 to continue the discussion about this. Certainly out of the scope of this PR. 💭